flash/image/classification/adapters.py

```diff
@@ -0,0 +1,518 @@
+
+
+if _LEARN2LEARN_AVAILABLE:
+    import learn2learn as l2l
+    from learn2learn.data.transforms import RemapLabels as Learn2LearnRemapLabels
+    from learn2learn.utils.lightning import Epochifier, TaskDataParallel
+else:
+
+    class Learn2LearnRemapLabels:
+        pass
+
+    class Epochifier:
+        pass
+
+    class TaskDataParallel:
+        pass
+
+
+class RemapLabels(Learn2LearnRemapLabels):
+    def remap(self, data, mapping):
+        # remap needs to be adapted to Flash API.
+        data[DefaultDataKeys.TARGET] = mapping(data[DefaultDataKeys.TARGET])
+        return data
+
+
+class NoModule:
+
+    """This class is used to prevent nn.Module infinite recursion."""
+
+    def __init__(self, task):
+        self.task = task
+
+    def __getattr__(self, key):
+        if key != "task":
+            return getattr(self.task, key)
+        return self.task
+
+    def __setattr__(self, key: str, value: Any) -> None:
+        if key == "task":
+            object.__setattr__(self, key, value)
+            return
+        setattr(self.task, key, value)
+
+
+class Model(torch.nn.Module):
+    def __init__(self, backbone: torch.nn.Module, head: Optional[torch.nn.Module]):
+        super().__init__()
+        self.backbone = backbone
+        self.head = head
+
+    def forward(self, x):
+        x = self.backbone(x)
+        if x.dim() == 4:
+            x = x.mean(-1).mean(-1)
+        if self.head is None:
+            return x
+        return self.head(x)
+
+
+class Learn2LearnAdapter(Adapter):
+
+    required_extras: str = "image"
+
+    def __init__(
+        self,
+        task: AdapterTask,
+        backbone: torch.nn.Module,
+        head: torch.nn.Module,
+        algorithm_cls: Type[LightningModule],
+        ways: int,
+        shots: int,
+        meta_batch_size: int,
+        queries: int = 1,
+        num_task: int = -1,
+        test_ways: Optional[int] = None,
+        test_shots: Optional[int] = None,
+        test_queries: Optional[int] = None,
+        default_transforms_fn: Optional[Callable] = None,
+        seed: int = 42,
+        **algorithm_kwargs,
+    ):
+        """The ``Learn2LearnAdapter`` is an :class:`~flash.core.adapter.Adapter` for integrating with `learn 2
+        learn` library (https://github.com/learnables/learn2learn).
+
+        Args:
+            task: Task to be used. This adapter should work with any Flash Classification task
+            backbone: Feature extractor to be used.
+            head: Predictive head.
+            algorithm_cls: Algorithm class coming
+                from: https://github.com/learnables/learn2learn/tree/master/learn2learn/algorithms/lightning
+            ways: Number of classes conserved for generating the task.
+            shots: The number of samples per label.
+            meta_batch_size: Number of task to be sampled and optimized over before doing a meta optimizer step.
+            queries: Number of support sample to be selected from the task.
```

Comment:
I was a bit confused here by the nomenclature of `shots` and `queries` but I confirm that it follows the l2l convention. Though, I'd like to try to summarize how I understand it in the current implementation

- `shots` is the number of samples used for adaptation
- `queries` is the number of samples used for validation
- `shots + queries` is the total number of samples per task

is this the correct interpretation?

Reply:
Yes, updated !